style_guide.md

```diff
@@ -0,0 +1,217 @@
+# Jsonnet Style Guide
+This style guide documents how we want to handle Jsonnet files when building the schemas with consistency. For more general principles and styling rules go to [Databricks repository](https://github.com/databricks/jsonnet-style-guide#databricks-jsonnet-guide).
+
+## Folder structure
+- The following structure is used:
+```
+england-wales/
+   ccs/
+      blocks/
+      lib/
+         rules.libsonnet
+   communal-establishment/
+   household/
+   individual/
+   lib/
+      common_rules.libsonnet
+      placeholders.libsonnet
+   ccs_household.jsonnet
+   census_communal_establishment.jsonnet
+   census_household.jsonnet
+   census_individual.jsonnet
+northern-ireland/
+
+```
+- England/Wales and Northern Ireland
+
+    - Jsonnet source files are separated into two folders: `england-wales` and `northern-ireland` with a number of sub-folders, each for corresponding schemas (e.g. `ccs`, `communal-establishment`, `household`, `individual`). These are divided into blocks in `blocks` folder.
+
+- Shared blocks
+
+    - Blocks from `individual` folder are used in both household and individual schema.
+
+- Top level Jsonnet files
+
+    - For each schema there are lists of block and placeholder imports.
+    - Schema structure is divided into sections, named after given schema and ending with `.jsonnet` suffix.
+
+- Lib folders
+
+    - Rules and placeholders are located in `lib` folder in corresponding schemas along with `blocks` folder.
+
+- Rules and placeholders
+
+    - These are separated into: `common_rules`, `placeholders` and `rules`. These files should end with `.libsonnet` suffix.
+
+- Top level and extended
+
+    - `common_rules` and `placeholders` are located in top-level "lib" folder.
+    - `rules` files for each schema are located in schema's sub-folder and extend some of the top-level `common_rules`.
+
+## Global variables
+
+- Top level arguments
+
+    - For each schema created it is possible to pass variables, e.g.: `region_code` and `census_month_year_date`, with different outputs being generated depending on their value.
+    - These are defined in eq-questionnaire-runner, are known at questionnaire launch and won't change during the Census.
+
+- External variables
+
+    - External variables are being used. Example of that is `census_month_year_date`.
+
+- Region Code
+
+    - Depending on `region_code` variable's value, different structure of sections is generated (e.g. certain questions omitted).
```

Comment:
- We should change `region_code` from a top-level argument to an external variable. This would mean it wouldn't have to be passed to blocks and would align with how we use `census_date`.
- The documentation should explain the difference and why one is used over another, something like:
  - Top-level arguments (`--tla-str`) are used to pass variables that are used in 5 blocks or less
  - External variables (`--ext-str`) are used to pass variables that are used in a more than 5 blocks

Reply:
This is all added, just moved around so it fits better.